File: binance4j-blvt/src/main/java/com/binance4j/blvt/param/TransactionRecordParams.java
package com.binance4j.blvt.param;

import com.binance4j.blvt.client.BLVTClient;
import com.binance4j.core.annotation.Param;
import com.binance4j.core.param.Params;

/**
 * {@link BLVTClient#getRedemptions} params.
 * 
 * @param tokenName Token name.
 * @param id        Subscription id.
 */
@Param
public record TransactionRecordParams(String tokenName, Long id) implements Params {
	/**
	 * Creates instance of the class.
	 * 
	 * @param tokenName Token name.
	 * @param id        Subscription id.
	 */
	public TransactionRecordParams(String tokenName, Long id) {
		this.id = id;
		this.tokenName = tokenName;
	}

	/**
	 * Creates instance of the class.
	 * 
	 * @param tokenName Token name.
	 */
	public TransactionRecordParams(String tokenName) {
		this(tokenName, null);
	}

	/**
	 * Creates instance of the class.
	 * 
	 * @param id Subscription id.
	 */
	public TransactionRecordParams(Long id) {
		this(null, id);
	}
}

File: binance4j-blvt/src/main/kotlin/com/binance4j/blvt/param/TransactionRecordParams.kt
package com.binance4j.blvt.param

import com.binance4j.blvt.client.BLVTClient
import com.binance4j.core.annotation.Param
import com.binance4j.core.param.Params

/**
 * [BLVTClient.getRedemptions] params.
 *
 * @property tokenName Token name.
 * @property id        Subscription id.
 */
@Param
data class TransactionRecordParams @JvmOverloads constructor(var tokenName: String? = null, var id: Long? = null) : Params